[docs] Add database maintenance section; update info message on ANALYZE run (sqlite) (#2756)
superseriousbusiness/gotosocial: docs/admin/database_maintenance.md

# Database Maintenance

Regardless of whether you choose to run GoToSocial with SQLite or Postgres, you may need to occasionally take maintenance steps to keep your database running well.

!!! tip
    

>     Though the maintenance tips provided here are intended by non-destructive, you should still consider backing up your database before manually intervening to do maintenance. That way if you mistype something or accidentally run a bad command, you can simply restore your backup and try again.

Reviewer: ```suggestion
    Though the maintenance tips provided here are intended to be non-destructive, you should backup your database before manually performing maintenance. That way if you mistype something or accidentally run a bad command, you can restore your backup and try again.
```

### Analyze / Optimize

Following [SQLite best practice](https://sqlite.org/lang_analyze.html#recommended_usage_pattern), GoToSocial runs the `optimize` SQLite pragma with `analysis_limit=1000` on closing database connections to keep index information up to date.

After each set of database migrations (eg., when starting a newer version of GoToSocial), GoToSocial will run `ANALYZE` to ensure that any indexes added or removed by migrations are taken into account correctly by the query planner.

The `ANALYZE` command may take ~10 minutes depending on your hardware and the size of your database file.

Because of the above automated steps, in normal circumstances you should not need to run manual `ANALYZE` commands against your SQLite database file.

However, if you interrupted a previous `ANALYZE` command, and you notice that queries are running remarkably slowly, it could be the case that the index metadata stored in SQLite's internal tables has been removed or undesirably altered.

If this is the case, you can try manually running a full `ANALYZE` command, by doing the following:

1. Stop GoToSocial.

> 2. While connected to your GoToSocial database file in the `sqlite3` shell, run `PRAGMA analysis_limit=0; ANALYZE;` (this may take quite a few minutes).

Reviewer: ```suggestion
2. While connected to your GoToSocial database file in the `sqlite3` shell, run `ANALYZE;` (this may take quite a few minutes).
```

There's not much point to `PRAGMA analysys_limit=0`, it's the same thing as not setting it at all.
